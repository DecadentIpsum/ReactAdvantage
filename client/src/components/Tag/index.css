.tag {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 8px;
  border-radius: 5px;
  color: #FFF; }
  .tag.announcement {
    background: #5A88C3; }
  .tag.enhancement {
    background: #EDC02F; }

.tag-icon {
  margin-right: 8px;
  font-size: 1rem; }

.tag-text {
  font-size: 1rem; }
