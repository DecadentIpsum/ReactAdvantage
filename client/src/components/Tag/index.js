import React, { Component } from 'react';
import './index.css';

export default class Tag extends Component {
    render() {
        const { type, text } = this.props;
        const classNames = ['tag', type];

        const iconClassNames = ['fa'];

        switch (type) {
            case 'announcement':
                iconClassNames.push('fa-bolt');
                break;
            case 'enhancement':
                iconClassNames.push('fa-bullhorn');
                break;
            default:
                break;
        }

        return (
            <div className={classNames.join(' ')}>
                <div className="tag-icon">
                    <i className={iconClassNames.join(' ')}></i>
                </div>
                <div className="tag-text">
                    {text}
                </div>
            </div>
        );
    }
}
