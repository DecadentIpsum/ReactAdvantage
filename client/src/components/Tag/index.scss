@import "global-styles/common.scss";

.tag {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 8px;
    border-radius: 5px;
    color: $white;

    &.announcement {
        background: $primary;
    }

    &.enhancement {
        background: $complementary;
    }
}

.tag-icon {
    margin-right: 8px;
    font-size: 1rem;
}

.tag-text {
    font-size: 1rem;
}
