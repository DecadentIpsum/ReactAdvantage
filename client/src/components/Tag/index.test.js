import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Tag from './index';

describe('tags renders correctly', () => {
    test('default Tag renders correctly', () => {
        const component = shallow(<Tag />);

        expect(component).toMatchSnapshot();
    });

    test('announcement Tag renders correctly', () => {
        const component = shallow(<Tag type="announcement" />);

        expect(component).toMatchSnapshot();
    });

    test('enhancement Tag renders correctly', () => {
        const component = shallow(<Tag type="enhancement" />);

        expect(component).toMatchSnapshot();
    });
});
