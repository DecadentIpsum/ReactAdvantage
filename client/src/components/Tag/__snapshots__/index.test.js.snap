// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags renders correctly announcement Tag renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    type="announcement"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tag-icon"
>
          <i
                    className="fa fa-bolt"
          />
</div>,
        <div
          className="tag-text"
/>,
      ],
      "className": "tag announcement",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-bolt"
/>,
          "className": "tag-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-bolt",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "tag-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tag-icon"
>
            <i
                        className="fa fa-bolt"
            />
</div>,
          <div
            className="tag-text"
/>,
        ],
        "className": "tag announcement",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-bolt"
/>,
            "className": "tag-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-bolt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "tag-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`tags renders correctly default Tag renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tag-icon"
>
          <i
                    className="fa"
          />
</div>,
        <div
          className="tag-text"
/>,
      ],
      "className": "tag ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa"
/>,
          "className": "tag-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "tag-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tag-icon"
>
            <i
                        className="fa"
            />
</div>,
          <div
            className="tag-text"
/>,
        ],
        "className": "tag ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa"
/>,
            "className": "tag-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "tag-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`tags renders correctly enhancement Tag renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    type="enhancement"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tag-icon"
>
          <i
                    className="fa fa-bullhorn"
          />
</div>,
        <div
          className="tag-text"
/>,
      ],
      "className": "tag enhancement",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-bullhorn"
/>,
          "className": "tag-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-bullhorn",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "tag-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tag-icon"
>
            <i
                        className="fa fa-bullhorn"
            />
</div>,
          <div
            className="tag-text"
/>,
        ],
        "className": "tag enhancement",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-bullhorn"
/>,
            "className": "tag-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-bullhorn",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "tag-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
